models/Asset/Video.php

```diff
@@ -216,9 +220,16 @@ public function getDuration(): float|int|null
 
     public function getDimensions(): ?array
     {
+        $dimensions = null;
         $width = $this->getCustomSetting('videoWidth');
         $height = $this->getCustomSetting('videoHeight');
-        if (!$width || !$height) {
+
+        if ($width && $height) {
+            $dimensions = [
```

Comment:
what about "reserving" the assignment of `$dimensions` to the output `$this->getDimensionsFromBackend();` 
```suggestion
            return [
```

remove `$dimensions = null;` on line 223 and have a  `return null` at the bottom (the implicit else all)

Reply:
But then we'd need another return `$dimensions` in the `getDimensionsFromBackend() / elseif` so nothing saved 😉